File: problems/reordered_power_of_2/solution.py
class Solution:
    def reorderedPowerOf2(self, n: int) -> bool:
        if n == 1:
            return True
        pows = list(range(1,30))
        for pow in pows:
            str1 = ''.join(sorted(str(2 ** pow)))
            str2 = ''.join(sorted(str(n)))
            if len(str1) > len(str2):
                break
            if str1 == str2:
                return True
        return False

File: problems/reordered_power_of_2/solution.js
const reorderedPowerOf2 = function(n) {
    if(n == 1) return true
    const pows = Array(29).fill(0).map((_,i) => i + 1)
    for(const pow of pows){
        let str1 = String(2 ** pow).split('').sort().join(''), str2 = String(n).split('').sort().join('')
        if(str1.length > str2.length) break
        if(str1 == str2) return true
    }
    return false
}